submission/submission-webapp/src/main/kotlin/ac/uk/ebi/biostd/submission/web/resources/MultipartSubmitResource.kt

```diff
@@ -131,17 +132,18 @@ class MultipartSubmitResource(
         @RequestParam(SUBMISSION) file: MultipartFile,
         @RequestParam(FILES) files: Array<MultipartFile>,
         @RequestParam(FILE_MODE, defaultValue = "COPY") mode: FileMode,
-        @RequestParam attributes: Map<String, String> = emptyMap()
+        @RequestParam(ATTRIBUTES) attributes: Array<ExtAttributeDetail>?
     ): Submission {
         val tempFiles = tempFileGenerator.asFiles(files)
         val subFile = tempFileGenerator.asFile(file)
+
         val contentWebRequest = FileSubmitWebRequest(
             submission = subFile,
             onBehalfRequest = onBehalfRequest,
             user = user,
             format = TSV,
             fileMode = mode,
-            attrs = attributes,
+            attrs = attributes?.associate { it.name to it.value } ?: emptyMap(),
```

Comment:
I think here you can do `attributes.orEmpty().associate { it.name to it.value }` not big change but lik 1% simpler.